### DISCUSSION TITLE
Multiple docker packagers ?

### DISCUSSION THREAD

**Discussion by gotson** [jreleaser/jreleaser#1513] [Q&A]
Is it possible to build multiple docker images with different configuration/dockerfile for the same app ? I tried having 2 `docker` blocks in the `packagers` configuration but it seems that doesn't work ?

**Comment by aalmiray** [comment]
Did you try using the `specs` option as explained in the docs? https://jreleaser.org/guide/latest/reference/packagers/docker.html#_multiple_dockerfile_per_distribution

JReleaser uses it to generate multiple images from different Dockerfiles

https://github.com/jreleaser/jreleaser/tree/main/src/jreleaser/distributions/jreleaser-standalone/docker
https://github.com/jreleaser/jreleaser/blob/2b387cb44078b522700c71e2f4ef0b43440b80b3/jreleaser.yml#L379-L409

**Reply by gotson** [reply]
I hadn't seen that, but I guess it don't work if there's a single artifact in the distribution, right?

**Reply by aalmiray** [reply]
I think you may configure matchers for extra properties. The example shows the explicit `platform` property.

**Reply by gotson** [reply]
I ended up using multi-stage docker builds using different stages per target architecture. That way they are all under the same manifest.
